app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt

```diff
@@ -5,157 +5,208 @@
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.activityViewModels
 import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModel
 import at.bitfire.icsdroid.R
-import at.bitfire.icsdroid.databinding.SubscriptionSettingsBinding
-import com.google.android.material.dialog.MaterialAlertDialogBuilder
-import com.google.android.material.switchmaterial.SwitchMaterial
-import org.joda.time.Minutes
-import org.joda.time.format.PeriodFormat
+import at.bitfire.icsdroid.ui.reusable.SwitchSetting
 
 class SubscriptionSettingsFragment : Fragment() {
 
     private val model by activityViewModels<SubscriptionSettingsModel>()
 
-    private lateinit var binding: SubscriptionSettingsBinding
-
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, inState: Bundle?): View {
-        binding = SubscriptionSettingsBinding.inflate(inflater, container, false)
-        binding.lifecycleOwner = this
-        binding.model = model
-
-        model.defaultAlarmMinutes.observe(
-            viewLifecycleOwner,
-            defaultAlarmObserver(
-                binding.defaultAlarmSwitch,
-                binding.defaultAlarmText,
-                model.defaultAlarmMinutes
-            )
-        )
-        model.defaultAllDayAlarmMinutes.observe(
-            viewLifecycleOwner,
-            defaultAlarmObserver(
-                binding.defaultAlarmAllDaySwitch,
-                binding.defaultAlarmAllDayText,
-                model.defaultAllDayAlarmMinutes
-            )
-        )
-
         val colorPickerContract = registerForActivityResult(ColorPickerActivity.Contract()) { color ->
             model.color.value = color
         }
-        binding.color.setOnClickListener {
-            colorPickerContract.launch(model.color.value)
+        return ComposeView(requireActivity()).apply {
+            setContent {
+                val url by model.url.observeAsState("")
+                val title by model.title.observeAsState("")
+                val color by model.color.observeAsState(0)
+                val ignoreAlerts by model.ignoreAlerts.observeAsState(false)
+                val defaultAlarmMinutes by model.defaultAlarmMinutes.observeAsState()
+                val defaultAllDayAlarmMinutes by model.defaultAllDayAlarmMinutes.observeAsState()
+                SubscriptionSettingsComposable(
+                    url = url,
+                    title = title,
+                    titleChanged = { model.title.postValue(it) },
+                    color = color,
+                    colorIconClicked = { colorPickerContract.launch(color) },
+                    ignoreAlerts = ignoreAlerts,
+                    ignoreAlertsChanged = { model.ignoreAlerts.postValue(it) },
+                    defaultAlarmMinutes = (defaultAlarmMinutes ?: "").toString(),
```

Comment:
Can't we just accept here `Int?`?

Reply:
yup, and if I remember correctly that's even best practice :+1: